src/components/Board.js

```diff
@@ -4,17 +4,20 @@ import Square from './Square';
 import PropTypes from 'prop-types';
 
 
-const generateSquareComponents = (squares, onClickCallback) => {
-  // Complete this for Wave 1
-  // squares is a 2D Array, but 
-  //  you need to return a 1D array
-  //  of square components
 
-}
+const generateSquareComponents = (squares, onClickCallback) => {
+    const flatArray = []
+    squares.forEach(row => {
+      row.forEach(square => {
+        flatArray.push(square)
+      })
+    })
+    const squaresArray = flatArray.map(squareData => <Square id={squareData.id} value={squareData.value} onClickCallback={onClickCallback}/>)
```

Comment:
There are some built in methods that can make this simpler.

You can use `flat` instead of your first loop:

```suggestion
    const flatArray = squares.flat();
    const squaresArray = flatArray.map(squareData => <Square id={squareData.id} value={squareData.value} onClickCallback={onClickCallback}/>)
```

And then there is actually a built-in method hat combines `flat` and `map`:

```suggestion
    const squaresArray = squares.flatMap(squareData => <Square id={squareData.id} value={squareData.value} onClickCallback={onClickCallback}/>)
```